src/mongo/db/repl/replication_coordinator_external_state_impl.cpp

```diff
@@ -607,17 +614,22 @@ Status ReplicationCoordinatorExternalStateImpl::storeLocalConfigDocument(Operati
                                                                          bool writeOplog) {
     try {
         writeConflictRetry(opCtx, "save replica set config", configCollectionName, [&] {
-            WriteUnitOfWork wuow(opCtx);
-            Lock::DBLock dbWriteLock(opCtx, configDatabaseName, MODE_X);
-            Helpers::putSingleton(opCtx, configCollectionName, config);
+            {
+                // Writes to 'local.system.replset' must be untimestamped.
+                WriteUnitOfWork wuow(opCtx);
+                Lock::DBLock dbWriteLock(opCtx, configDatabaseName, MODE_X);
+                Helpers::putSingleton(opCtx, configCollectionName, config);
+                wuow.commit();
+            }
 
             if (writeOplog) {
+                WriteUnitOfWork wuow(opCtx);
```

Comment:
Can we add a comment explaining that it's okay to separate the noop from the write to local.system.replset since the noop doesn't affect correctness of the protocol?

Reply:
Done.